File: solutions/python/clock/1/clock.py
class Clock:
    def __init__(self, hour, minute):
        total_minutes = (hour * 60 + minute) % 1440
        if total_minutes < 0:
            total_minutes += 1440
        self.hour = total_minutes // 60
        self.minute = total_minutes % 60

    def __repr__(self):
        return f"Clock({self.hour}, {self.minute})"

    def __str__(self):
        return f"{self.hour:02}:{self.minute:02}"

    def __eq__(self, other):
        if not isinstance(other, Clock):
            return False
        return self.hour == other.hour and self.minute == other.minute

    def __add__(self, minutes):
        return Clock(self.hour, self.minute + minutes)

    def __sub__(self, minutes):
        return Clock(self.hour, self.minute - minutes)

File: solutions/cpp/clock/1/clock.cpp
#include "clock.h"

#include <iomanip>
#include <sstream>

namespace date_independent {
    clock::clock(const int hours, const int minutes) {
        int total_minutes = (hours * 60 + minutes) % 1440;

        if (total_minutes < 0) {
            total_minutes += 1440;
        }

        hours_ = total_minutes / 60;
        minutes_ = total_minutes % 60;
    }

    clock clock::at(const int hours, const int minutes) {
        return {hours, minutes};
    }

    clock clock::plus(const int minutes) const {
        return {hours_, minutes_ + minutes};
    }

    bool clock::operator==(const clock &other) const {
        return minutes_ == other.minutes_ && hours_ == other.hours_;
    }

    bool clock::operator!=(const clock &other) const {
        return !(*this == other);
    }

    clock::operator std::string() const {
        std::ostringstream oss;

        oss << std::setw(2)
                << std::setfill('0')
                << hours_
                << ":"
                << std::setw(2)
                << std::setfill('0')
                << minutes_;

        return oss.str();
    }
} // namespace date_independent
